torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1822,60 +1822,61 @@ def _load_local_state_dict(
             return self.load_state_dict(state_dict, *args)
 
     def forward(self, *args: Any, **kwargs: Any) -> Any:
-        self._lazy_init()
+        with torch.autograd.profiler.record_function("FullyShardedDataParallel.forward"):
```

Comment:
would I be correct that this will be a noop outside of a profiler context? And will this incur any overhead?

Reply:
Yep, should be a noop when profiler is not enabled. Will run benchmarks to gauge overhead.